### ISSUE TITLE
markOthers aria-live exclusion does not work in Safari + VoiceOver with aria-modal

### ISSUE THREAD

**Issue by mihkeleidast** [floating-ui/floating-ui#3359]
**Describe the bug**

https://github.com/floating-ui/floating-ui/blob/master/packages/react/src/utils/markOthers.ts#L163

Currently, the markOthers utility function excludes aria-live etc et elements from being marked as hidden, so they would still be available to assistive tech. It works in most browsers and screen readers, but not in Safari + VoiceOver when for example there's a modal dialog open on the page.

See the following issues about this topic:
- https://github.com/w3c/aria/issues/1854
- https://bugs.webkit.org/show_bug.cgi?id=276553

**To Reproduce**

1. Turn on VoiceOver
2. Go to https://codesandbox.io/p/sandbox/ancient-forest-s2qnp4?file=%2Fsrc%2FApp.tsx%3A19%2C44
3. Click on "my trigger" button
4. Click on "increase" button
5. Observe that the updated aria-live region content is not announced by VoiceOver

You can also observe in Safari dev tools that the live region is marked as "ignored" once the dialog with aria-modal is open.

**Screenshots**

Can record a video of the above behavior if you like.

**Context:**

- OS: macOS 15.5
- Browser: Safari 18.5

**Additional context**

You've probably gathered that this is not a bug that can be fixed in floating-ui per se, but I think the current behavior should be changed so it would work the same across browsers and assistive tech, even if it sort of means breaking it for other assistive tech for now. I think the current exclusion method may do more harm than good - if it works with most browsers and screen reader combos, devs can forget to check VoiceOver + Safari in an aria-modal. If the live regions need a workaround for that combo, wouldn't it be better if we did not have the exclusion here, forcing devs to think about this earlier on?

How it should work with modals is a bit undefined logic based on that aria issue, but the general idea with a modal dialog is that everything else is not available. The exclusion in markOthers tries to fight against that logical assumption, I think.

**Comment by mihkeleidast** [comment]
btw JAWS does announce dialogs with aria-modal differently, explicitly says "modal dialog" instead of "dialog", but as that's not the case anywhere alse, I think it's not too important.

anyway, thanks for the thoughts, removing aria-modal from our components now as the floating-ui method is functionally equivalent otherwise.
